### ISSUE TITLE
Kinemage least-significant-digit script fails on 6qt0 output

### ISSUE THREAD

**Issue by russell-taylor** [ReliaSolve/cctbx_project#103]
Please fix this in the master branch of the suitename_regression project, where the script is named filter_kinemage_diff_low_order_bit.py.

The script seems to produce a false positive:

```
taylorr@reliasolve-3:~/src/suitename_regression/outputs$ diff 6qt0.kinemage.orig.cleaned 6qt0.kinemage.new.cleaned | python3 ../filter_kinemage_diff_low_order_bit.py 
948c948
< {33 p 1a :D==0.429:S==0.611:  :1: A:1732: : :  U}  ,  0.000, 80.869,220.466,302.641,281.998,169.837, 65.627, 77.812,  0.000
---
> {33 p 1a :D==0.428:S==0.611:  :1: A:1732: : :  U}  ,  0.000, 80.869,220.466,302.641,281.998,169.837, 65.627, 77.812,  0.000
```

Of interest, if it is run on the non-cleaned file, it crashes:

```
taylorr@reliasolve-3:~/src/suitename_regression/outputs$ diff 6qt0.kinemage.orig 6qt0.kinemage.new | python3 ../filter_kinemage_diff_low_order_bit.py 
Traceback (most recent call last):
  File "../filter_kinemage_diff_low_order_bit.py", line 37, in <module>
    main()
  File "../filter_kinemage_diff_low_order_bit.py", line 20, in main
    before1, a, after1 = m1.group(1, 2, 3)
AttributeError: 'NoneType' object has no attribute 'group'
```

The crash is not a problem so long as it doesn't expose other cases where it will crash, but I've set up the script to make these into false positives so they won't slip by us.

**Comment by russell-taylor** [comment]
[files.zip](https://github.com/ReliaSolve/cctbx_project/files/6902033/files.zip)
Here are the relevant files to test on

**Comment by russell-taylor** [comment]
This is also failing on 6hhq, with differences in the D= and S= portions (see below).  Looks like the diff script is not properly ignoring these low-order-digit differences.

```
142c142
< {33 p 1a :D==0.296:S==0.800:  :1: 1: 130: : :  A}  ,  0.000, 76.506,218.927,275.576,289.653,168.858, 50.509, 77.216,  0.000
---
> {33 p 1a :D==0.296:S==0.799:  :1: 1: 130: : :  A}  ,  0.000, 76.506,218.927,275.576,289.653,168.858, 50.509, 77.216,  0.000
2212c2212
< {33 p 1a :D==0.441:S==0.593:  :1: A: 726: : :  C}  ,  0.000, 76.585,235.686,292.343,300.867,157.789, 59.950, 82.212,  0.000
---
> {33 p 1a :D==0.440:S==0.593:  :1: A: 726: : :  C}  ,  0.000, 76.585,235.686,292.343,300.867,157.789, 59.950, 82.212,  0.000
3065c3065
< {33 p 1a :D==0.535:S==0.446:  :1:AR: 576: : :  C}  ,  0.000, 76.119,210.058,281.359,280.223,175.257, 70.488, 76.479,  0.000
---
> {33 p 1a :D==0.535:S==0.445:  :1:AR: 576: : :  C}  ,  0.000, 76.119,210.058,281.359,280.223,175.257, 70.488, 76.479,  0.000
4398c4398
< {33 p 1a :D==0.480:S==0.531:  :1:AR:3262: : :  U}  ,  0.000, 80.477,213.047,287.160,300.986,187.867, 37.241, 82.516,  0.000
---
> {33 p 1a :D==0.481:S==0.531:  :1:AR:3262: : :  U}  ,  0.000, 80.477,213.047,287.160,300.986,187.867, 37.241, 82.516,  0.000
4615c4615
< {33 p 1a :D==0.404:S==0.649:  :1:AT: 153: : :  U}  ,  0.000, 81.155,233.445,281.917,306.352,160.532, 50.091, 80.908,  0.000
---
> {33 p 1a :D==0.404:S==0.648:  :1:AT: 153: : :  U}  ,  0.000, 81.155,233.445,281.917,306.352,160.532, 50.091, 80.908,  0.000
4724c4724
< {33 p 1a :D==0.831:S==0.069:  :1:sR: 239: : :  C}  ,  0.000, 74.221,209.691,265.239,322.126,165.554, 30.437, 72.513,  0.000
---
> {33 p 1a :D==0.831:S==0.068:  :1:sR: 239: : :  C}  ,  0.000, 74.221,209.691,265.239,322.126,165.554, 30.437, 72.513,  0.000
6849c6849
< {33 t 1c :D==0.564:S==0.400:  :1:sR: 998: : :  A}  ,  0.000, 80.737,189.641,283.716,160.344,216.652,161.274, 77.809,  0.000
---
> {33 t 1c :D==0.564:S==0.399:  :1:sR: 998: : :  A}  ,  0.000, 80.737,189.641,283.716,160.344,216.652,161.274, 77.809,  0.000

```

**Comment by russell-taylor** [comment]
We have many dozen Kinemage failures in the whole run, so I'm going to try and figure out what the script is doing and how to fix it rather than trying to look through all of the diffs one at a time both this time and in the future.

**Comment by russell-taylor** [comment]
Fixed in commit 1738332fb0d0aeb7b3b9159e00521d74bc0058ff to both not crash when there is not a match and to check both S== and D== entries.

**Comment by Joymaker** [comment]
Commit 1738332fb0d0aeb7b3b9159e00521d74bc0058ff of what repository? Pulled cctbx-project new where I originally committed, and I don't see your commit.
On 7/30/2021 7:49:19 PM, Russell Taylor ***@***.***> wrote:
Fixed in commit 1738332fb0d0aeb7b3b9159e00521d74bc0058ff to both not crash when there is not a match and to check both S== and D== entries.

**Comment by russell-taylor** [comment]
https://github.com/ReliaSolve/suitename_regression is the repository with that commit.  This was a fix on the error script.
